cloudinit/config/cc_kernel_modules.py

```diff
@@ -0,0 +1,370 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""Kernel Modules"""
+import re
+from array import array
+from logging import Logger
+from textwrap import dedent
+from typing import List
+
+from cloudinit import log as logging
+from cloudinit import subp, util
+from cloudinit.cloud import Cloud
+from cloudinit.config import Config
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.settings import PER_INSTANCE
+
+MODULE_DESCRIPTION = dedent(
+    """\
+Manages and enhances kernel modules on a systemd-based system.
+This module is capable of loading kernel modules at boot as well as
+enhancing it with parameters.
+Beside applying settings during runtime it will also persist all
+settings in ``/etc/modules-load.d`` and ``/etc/modprobe.d``.
+"""
+)
+
+DISTROS = ["debian", "ubuntu"]
+
+meta: MetaSchema = {
+    "id": "cc_kernel_modules",
+    "name": "Kernel Modules",
+    "title": "Manage and enhance kernel modules",
+    "description": MODULE_DESCRIPTION,
+    "distros": DISTROS,
+    "frequency": PER_INSTANCE,
+    "activate_by_schema_keys": ["kernel_modules"],
+    "examples": [
+        dedent(
+            """\
+    kernel_modules:
```

Comment:
Please add other viable examples using various persist keys beyond `options` to show folks how to use such config keys,

We also need happy path unittest coverage of each key to make sure it's rendered properly to config files.